### PR TITLE
fix: REPL completion distinguishes class-side vs instance methods

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

- **Bug 1:** `Counter in<TAB>` was showing instance methods (`increment`, `inspect`, `instVarAt:`, etc.) instead of class-side methods (`spawn`, `new`, `subclass:`, etc.). Class-name receivers now correctly complete class-side methods.
- **Bug 2:** `c in<TAB>` (where `c` is bound to a `Counter` actor) returned nothing. Variable bindings are now looked up in the session state and their `#beamtalk_object{class}` is used to provide instance method completions.

## Changes

- `beamtalk_object_class`: export `local_class_methods/1` wrapping existing `get_local_class_methods` gen_server call
- `beamtalk_repl_ops_dev`: replace `receiver_to_class_name/1` with `classify_receiver/2` that returns `{class, Name} | {instance, Name} | undefined`; add `collect_all_class_methods/2`, `builtin_class_methods/0`, `get_session_bindings/1`; thread `SessionPid` and bindings through the complete handler
- `beamtalk_repl_server_tests`: 3 new unit tests for binding lookup paths

## Test plan

- [x] `just test` passes (4,800+ tests)
- [x] Pre-push hook passes (clippy, dialyzer, erlfmt, biome)
- [ ] Manual REPL verification: `Counter sp<TAB>` → shows `spawn`, `spawnWith:`, `subclass:`
- [ ] Manual REPL verification: `c := Counter spawn; c in<TAB>` → shows `increment`, `inspect`, etc.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * New API to query local (non-inherited) class-level methods from class objects
  * Enhanced code completion that leverages current session context when suggesting available methods for both classes and instances
  
* **Tests**
  * Expanded test coverage for session-aware code completions with various binding scenarios
  * Added tests for symbol information queries and edge cases

### CODE REVIEW

File: runtime/apps/beamtalk_workspace/src/beamtalk_repl_ops_dev.erl [hunk 1]
```diff
@@ -486,6 +527,60 @@ maybe_class(ClassName) ->
...
+    end.
+
+%% @private
+%% @doc The built-in class-side methods available on every concrete class.
+-spec builtin_class_methods() -> [atom()].
+builtin_class_methods() ->
+    [spawn, new, 'new:', 'spawnWith:', 'subclass:'].
+
+%% @private
+%% @doc Collect all class-side method selectors for a class by walking the superclass chain.
```
[the remark this answers is not in the document]
Comment (jamesc) [reply runtime/apps/beamtalk_workspace/src/beamtalk_repl_ops_dev.erl]: Added the missing depth guard doc comment to `collect_all_class_methods` to match the pattern from `collect_all_methods`. Fixed in ad502b4.
